lib/IMAP/PreviewEnhancer.php

```diff
@@ -34,14 +35,24 @@ class PreviewEnhancer {
 	/** @var LoggerInterface */
 	private $logger;
 
+	/** @var AvatarService */
+	private $avatarService;
+
+	/** @var string */
+	private $UserId;
+
 	public function __construct(IMAPClientFactory $clientFactory,
 		ImapMapper $imapMapper,
 		DbMapper $dbMapper,
-		LoggerInterface $logger) {
+		LoggerInterface $logger,
+		AvatarService $avatarService,
+		string $UserId) {
```

Comment:
```suggestion
		string $userId) {
```

https://docs.nextcloud.com/server/latest/developer_manual/basics/dependency_injection.html#predefined-core-services